pca_cca.go

```diff
@@ -0,0 +1,252 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"errors"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// PrincipalComponents returns the principal component direction vectors and
```

Comment:
Yep looks good, I left the comment
`// PrincipalComponents centers the variables but does not scale the variance.`
though as this seems relevant.